client/state/user-settings/thunks/create-user-email-pending-update.js

```diff
@@ -0,0 +1,46 @@
+import { translate } from 'i18n-calypso';
+import wp from 'calypso/lib/wp';
+import { errorNotice, infoNotice } from 'calypso/state/notices/actions';
+import 'calypso/state/user-settings/init';
+
+/**
+ * Redux thunk which exclusively updates the `email` setting.
+ * @param {string} newEmail The new email address
+ */
+export const createUserEmailPendingUpdate = ( newEmail ) => async ( dispatch ) => {
+	try {
+		const { user_email_change_pending, new_user_email } = await wp
+			.me()
+			.settings()
+			.update( { user_email: newEmail } );
+
+		if ( user_email_change_pending ) {
+			dispatch(
+				infoNotice(
+					translate(
+						'There is a pending change of your WordPress.com email to %(newEmail)s. Please check your inbox for a confirmation link.',
+						{
+							args: { newEmail: new_user_email },
+						}
+					),
+					{
+						showDismiss: true,
+						isPersistent: true,
+					}
+				)
+			);
+		}
+	} catch ( error ) {
+		dispatch(
+			errorNotice(
+				translate( 'There was a problem updating your WordPress.com Account email: %(error)s', {
```

Comment:
I notice that we weren't previously displaying the error returned from the server. I initially got an error message saying "A fresh access token must be used to query information about the current user." I'm not sure whether that extra message returned by the API that useful for a user. How would I refresh an access token?

Reply:
In this case it's not helpful, that's true. But there are cases where the e-mail is already been used, and that case is returned by the API. Maybe if there's a way to get it through a constant, we could display this specific error.